wgpu-hal/src/vulkan/device.rs

```diff
@@ -535,6 +535,40 @@ impl super::Device {
             config: config.clone(),
         })
     }
+
+    pub fn image_raw_flags(desc: &crate::TextureDescriptor) -> vk::ImageCreateFlags {
+        let mut raw_flags = vk::ImageCreateFlags::empty();
+        if desc.dimension == wgt::TextureDimension::D2 && desc.size.depth_or_array_layers % 6 == 0 {
```

Comment:
this is a dirty hack, but we still need to do our best
let's also check for width == height here